contentful/marketplace-partner-apps
# feat: add Rich Text Toolkit app

## Purpose

Initial commit for the Rich Text Toolkit app. This app allows content editors to import HTML and Google Doc content into rich text fields.

## Approach

The project was setup using the Contentful CLI bootstrapper [create-contentful-app](https://github.com/contentful/create-contentful-app)

This application extends the Contentful [Rich Text Field Editor](https://github.com/contentful/field-editors/tree/master/packages/rich-text) to add additional functionality.

Added functionality:

1. Import to rich text fields - A button is added to the rich text field editor which allows users to paste HTML or Google document content. Our API converts this content from its’ source format to the Contentful rich text format, including image assets and complex content structures.
![import](https://github.com/user-attachments/assets/93c8dbc9-7020-4fce-9a56-8330416213d3)


2. JSON preview - A preview tab is added to the rich text editor field, allowing content authors to preview or copy the underlying JSON data.
![json_preview](https://github.com/user-attachments/assets/ba3900cc-041f-4f0a-801f-2dfab83c8b04)



A demo can be found [here](https://ellavationlabs.com/richtext-toolkit/#demo), along with additional information.

The configuration page allows these functions and details around their configuration to be enabled/disabled to meet the clients’ needs.


## Testing steps

1. Install application to space
2. Open app configuration page. 
3. Confirm page renders properly
4. Enable both Import and JSON preview features
5. Open a content model with a rich text editor field
6. Set Rich Text Toolkit as the appearance editor for the desired rich text field
7. Edit a content entry belonging to the above edited model
8. Confirm Rich text content entry now has JSON preview tab, and that tab populates JSON content when selected
 Example screenshot
9. Press import button and paste sample HTML below:
```
<!DOCTYPE html>
<html>
<body>
    <h1>My First Heading</h1>
    <p>My first paragraph.</p>
    <ul>
        <li>Apples</li>
        <li>Oranges</li>
        <li>Pears</li>
    </ul>
    <img src="https://images.ctfassets.net/jtqsy5pye0zd/6wNuQ2xMvbw134rccObi0q/bf61badc6d6d9780609e541713f0bba6/Contentful_Logo_2.5_Dark.svg"
        alt="test image" />
</body>
</html>
```
10. Confirm Rich text editor populates with content as shown below
![test result](https://github.com/user-attachments/assets/8104b3a5-2b1e-40c4-90d0-8302f835b99a)


## Breaking Changes
N/A

## Dependencies and/or References
https://ellavationlabs.com/richtext-toolkit

## Deployment

Locations:

- App configuration Screen
- Entry field
    - Rich text
